### ISSUE TITLE
Issue with `export class Bug { ... }; export default Bug;` in Babel 7

### ISSUE THREAD

**Issue by Ashoat** [facebook/metro#199]
I am seeing this issue with `react-native@0.56.0` and `metro@0.38.3` (and also `metro@0.38.1`).

Here's the code that causes the issue:

```js
export class Bug {

  // To get the same error we're seeing, this static declaration is necessary.
  // However, the error we're seeing is likely hiding another one, because if
  // this declaration is removed, another error occurs.
  static blah = {};

  constructor(props) {
    this.test(props);
  }

  test = (props) => {
    // This is the line that is actually causing the real error
    const test = props.test;
  }
}

export default Bug;
```

I've created a repo with this minimal repro [here](https://github.com/Ashoat/rn56babelbug). I attempted to repro the issue without Metro, by setting up a Babel project with the same code and `babel-preset-react-native` [here](https://github.com/Ashoat/babel7bug), but was unable to repro. My conclusion is that the issue has something to do with how Metro consumes Babel.

**Comment by Ashoat** [comment]
I initially found this problem while investigating [this React Native](https://github.com/facebook/react-native/issues/20038) issue. Additionally I think [this React Native](https://github.com/facebook/react-native/issues/20153) issue might be related, which @charpeni was able to [repro](https://github.com/charpeni/metro-static-properties-issue) on a naked Metro setup.

In both cases, the issue can be addressed by using two separate expressions to declare the class versus export it.

**Comment by Ashoat** [comment]
[This Babel issue](https://github.com/babel/babel/issues/7177) may be related. It looks like the specific issue there had to do with [plugin ordering](https://github.com/babel/babel/issues/7177#issuecomment-365952402), which perhaps is the culprit here (though I doubt in the same way).

**Comment by rafeca** [comment]
@mjesun does this have anything to do with the Babel issues that manifest on our constant folding plugin? ([more info](https://github.com/facebook/metro/issues/208))

**Comment by mjesun** [comment]
Yeah, it could be. Apparently scope traversing has some bugs in Babel. I've tried pinging them with no luck, though.

**Comment by Ashoat** [comment]
If we could get a pure-Babel repro I think it would easier to get their attention. But [my attempts](https://github.com/Ashoat/babel7bug) at that were unsuccessful, probably because I’m not accurately mirroring the way Metro consumes Babel. Any advice?

**Comment by mjesun** [comment]
The issues I've found so far are related to calling `path.scope.crawl()` after performing some evaluations. See an example in https://github.com/babel/babel/issues/8387; maybe this helps repro'ing.

**Comment by kelset** [comment]
I think this may be fixed via Babel 7.0 and Metro 0.44.x (once this is merged https://github.com/facebook/metro/pull/233)
